xml/System.Text.Json.Serialization/ReferenceHandler.xml

```diff
@@ -73,9 +73,38 @@
         <ReturnType>System.Text.Json.Serialization.ReferenceHandler</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Metadata properties will be honored when deserializing JSON objects and arrays into reference types and written when serializing reference types. This is necessary to create round-trippable JSON from objects that contain cycles or duplicate references.</summary>
         <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+* On Serialize:
+ When writing complex reference types, the serializer also writes metadata properties (`$id`, `$values`, and `$ref`) within them.
+ The output JSON will contain an extra `$id` property for every object, and for every enumerable type the JSON array emitted will be nested within a JSON object containing an `$id` and `$values` property.
+ <xref:System.Object.ReferenceEquals(System.Object,System.Object)> is used to determine whether objects are identical.
+ When an object is identical to a previously serialized one, a pointer (`$ref`) to the identifier (`$id`) of such object is written instead.
+ No metadata properties are written for value types.
+ * On Deserialize:
+ The metadata properties within the JSON that are used to preserve duplicated references and cycles will be honored as long as they are well-formed**.
+ For JSON objects that don't contain any metadata properties, the deserialization behavior is identical to `null`.
+ For value types:
+ * The `$id` metadata property is ignored.
+ * A <xref:System.Text.Json.JsonException> is thrown if a `$ref` metadata property is found within the JSON object.
+ * For enumerable value types, the `$values` metadata property is ignored.
+ ** For the metadata properties within the JSON to be considered well-formed, they must follow these rules:
+ 1) The `$id` metadata property must be the first property in the JSON object.
+ 2) A JSON object that contains a `$ref` metadata property must not contain any other properties.
+ 3) The value of the `$ref` metadata property must refer to an `$id` that has appeared earlier in the JSON.
+ 4) The value of the `$id` and `$ref` metadata properties must be a JSON string.
+ 5) For enumerable types, such as <xref:System.Collections.Generic.List`1>, the JSON array must be nested within a JSON object containing an `$id` and `$values` metadata property, in that order.
+ 6) For enumerable types, the `$values` metadata property must be a JSON array.
+ 7) The `$values` metadata property is only valid when referring to enumerable types.
+ If the JSON is not well-formed, a <xref:System.Text.Json.JsonException> is thrown.
```

Comment:
```suggestion
* On Serialize:

  * When writing complex reference types, the serializer also writes metadata properties (`$id`, `$values`, and `$ref`) within them.
  * The output JSON will contain an extra `$id` property for every object, and for every enumerable type the JSON array emitted will be nested within a JSON object containing an `$id` and `$values` property.
  * <xref:System.Object.ReferenceEquals(System.Object,System.Object)> is used to determine whether objects are identical.
  * When an object is identical to a previously serialized one, a pointer (`$ref`) to the identifier (`$id`) of such object is written instead.
  * No metadata properties are written for value types.

* On Deserialize:

  * The metadata properties within the JSON that are used to preserve duplicated references and cycles will be honored as long as they are well-formed.\*
  * For JSON objects that don't contain any metadata properties, the deserialization behavior is identical to `null`.
  * For value types, the `$id` metadata property is ignored. A <xref:System.Text.Json.JsonException> is thrown if a `$ref` metadata property is found within the JSON object.
  * For enumerable value types, the `$values` metadata property is ignored.
 
\* For the metadata properties within the JSON to be considered well-formed, they must follow these rules:

* The `$id` metadata property must be the first property in the JSON object.
* A JSON object that contains a `$ref` metadata property must not contain any other properties.
* The value of the `$ref` metadata property must refer to an `$id` that has appeared earlier in the JSON.
* The value of the `$id` and `$ref` metadata properties must be a JSON string.
* For enumerable types, such as <xref:System.Collections.Generic.List%60>, the JSON array must be nested within a JSON object containing an `$id` and `$values` metadata property, in that order.
* For enumerable types, the `$values` metadata property must be a JSON array.
* The `$values` metadata property is only valid when referring to enumerable types.

If the JSON is not well-formed, a <xref:System.Text.Json.JsonException> is thrown.
```

Reply:
This looks great!
Just one tweak:
> For JSON objects that don't contain any metadata properties, the deserialization behavior is identical to `null`.

This sentence doesn't make too much sense, before it used to say:
> For JSON objects that don't contain any metadata properties, the deserialization behavior is identical to \<see cref="ReferenceHandling.Default"/>

But we removed `ReferenceHandler.Default`.

I would say instead:
> For JSON objects that don't contain any metadata properties, the deserialization behavior is identical to not using `Preserve`.

Reply:
I applied this change to my suggestion.